cmd/antrea-controller/options_test.go

```diff
@@ -0,0 +1,178 @@
+				ClusterCIDRs:         []string{"10.10.0.0/16", "a:b::0/64", "20.20.0.0/24"},
+				ServiceCIDR:          "172.16.0.0/16",
+				ServiceCIDRv6:        "2620:124::0/64",
+				NodeCIDRMaskSizeIPv4: 24,
+				NodeCIDRMaskSizeIPv6: 80,
+			},
+			expectedErr: "at most two cluster CIDRs may be specified",
+		},
+		{
+			name: "at most one cluster CIDR of each type",
+			nodeIPAMConfig: controllerconfig.NodeIPAMConfig{
+				EnableNodeIPAM:       true,
+				ClusterCIDRs:         []string{"10.10.0.0/16", "20.20.0.0/24"},
+				ServiceCIDR:          "172.16.0.0/16",
+				ServiceCIDRv6:        "2620:124::0/64",
+				NodeCIDRMaskSizeIPv4: 24,
+				NodeCIDRMaskSizeIPv6: 80,
+			},
+			expectedErr: "at most one cluster CIDR may be specified for each IP family",
+		},
+		{
+			// 	ipamIPv4MaskLo is 16, ipamIPv4MaskHi is 30
+			name: "invalid node IPv4 CIDR mask size",
+			nodeIPAMConfig: controllerconfig.NodeIPAMConfig{
+				EnableNodeIPAM:       true,
+				ClusterCIDRs:         []string{"10.10.0.0/16", "a:b::0/64"},
+				ServiceCIDR:          "172.16.0.0/16",
+				ServiceCIDRv6:        "2620:124::0/64",
+				NodeCIDRMaskSizeIPv4: 15,
+				NodeCIDRMaskSizeIPv6: 80,
+			},
+			expectedErr: "node IPv4 CIDR mask size 15 is invalid, should be between 16 and 30",
+		},
+		{
+			//	ipamIPv6MaskLo is 64, ipamIPv6MaskHi is 126
+			name: "invalid node IPv6 CIDR mask size",
+			nodeIPAMConfig: controllerconfig.NodeIPAMConfig{
+				EnableNodeIPAM:       true,
+				ClusterCIDRs:         []string{"10.10.0.0/16", "a:b::0/64"},
+				ServiceCIDR:          "172.16.0.0/16",
+				ServiceCIDRv6:        "2620:124::0/64",
+				NodeCIDRMaskSizeIPv4: 24,
+				NodeCIDRMaskSizeIPv6: 63,
+			},
+			expectedErr: "node IPv6 CIDR mask size 63 is invalid, should be between 64 and 126",
+		},
+		{
+			name: "invalid service CIDR",
+			nodeIPAMConfig: controllerconfig.NodeIPAMConfig{
+				EnableNodeIPAM:       true,
+				ClusterCIDRs:         []string{"10.10.0.0/16", "a:b::0/64"},
+				ServiceCIDR:          "1",
+				ServiceCIDRv6:        "2620:124::0/64",
+				NodeCIDRMaskSizeIPv4: 24,
+				NodeCIDRMaskSizeIPv6: 80,
+			},
+			expectedErr: "service CIDR 1 is invalid",
+		},
+		{
+			name: "invalid secondary service CIDR",
+			nodeIPAMConfig: controllerconfig.NodeIPAMConfig{
+				EnableNodeIPAM:       true,
+				ClusterCIDRs:         []string{"10.10.0.0/16", "a:b::0/64"},
+				ServiceCIDR:          "172.16.0.0/16",
+				ServiceCIDRv6:        "2",
+				NodeCIDRMaskSizeIPv4: 24,
+				NodeCIDRMaskSizeIPv6: 80,
+			},
+			expectedErr: "secondary service CIDR 2 is invalid",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			o := &Options{config: &controllerconfig.ControllerConfig{NodeIPAM: tc.nodeIPAMConfig}}
+			err := o.validateNodeIPAMControllerOptions()
+			if tc.expectedErr == "" {
+				assert.NoError(t, err)
+			} else {
+				assert.ErrorContains(t, err, tc.expectedErr, "validate NodeIPAMControllerOptions error is not as expected")
```

Comment:
```suggestion
				assert.ErrorContains(t, err, tc.expectedErr, "got error when validating NodeIPAMControllerOptions")
```
or just remove the extra message, I feel it's not necessary.